packages/starlight/integrations/shared/pathToLocale.ts

```diff
@@ -17,7 +17,7 @@ export function pathToLocale(
 		astroConfig,
 	}: {
 		starlightConfig: { locales: StarlightConfig['locales'] };
-		astroConfig: { root: URL; srcDir: URL };
+		astroConfig: { root: URL | string; srcDir: URL };
```

Comment:
```suggestion
		astroConfig: { root: AstroConfig['root']; srcDir: AstroConfig['srcDir'] };
```

I guess a more appropriate fix would be something similar to what is done for [`AsidesOptions`](https://github.com/withastro/starlight/blob/main/packages/starlight/integrations/asides.ts#L23) and reference the Astro types (with `AstroConfig` being from `import type { AstroConfig } from 'astro';`).

Altho, this would not fix the issue as it's typed as `URL` in Astro. I think the issue is located in Expressive Code which [defines it](https://github.com/expressive-code/expressive-code/blob/main/packages/astro-expressive-code/src/astro-config.ts#L26) as `URL | string`. I guess a fix would need to happen there too (and maybe refactor this entire type to rely on Astro's types, cherry pick what is needed, make some bits partial or use a deep partial type helper, etc. instead of duplicating them manually?).

cc @hippotastic

Reply:
If there's a way where we can make the type more accurate and not break in the future, I'm all for it!
I am personally not super familiar with astro/starlight types, nor zod, so I don't have a strong opinion here.

Reply:
> It might make sense to change your publishing strategy in light of this

Definitely, there is this [discussion](https://github.com/withastro/starlight/discussions/983) summarizing the issue but not something that has been tackled yet.

> If there's a way where we can make the type more accurate and not break in the future, I'm all for it!

Yeah, I think the change you suggested (with the above suggestion) and a fix on the Expressive Code side (once confirmed by Hippo) would be the best way to go forward.

Reply:
Just FYI, the type has been fixed in Expressive Code v0.34.0.

Reply:
> Just FYI, the type has been fixed in Expressive Code v0.34.0.

Amazing 🙌 

I guess applying the two suggested changes + the EC update would then fix the issues.

Reply:
Done :) amended my commit and pushed

Reply:
Thanks a lot :raised_hands: If you dont't mind, I think we should remove the EC bump from this PR as not strictly necessary for this PR and it also looks like there is another [PR](https://github.com/withastro/starlight/pull/1720) opened for this as it looks it may requires extra changes and may contain some breaking changes by quickly skimming through it.

Reply:
Makes sense, didn't think of the fact that pre-v1.0.0 minor updates would usually contain breaking changes